joschahenningsen/TUM-Live
# Search Box to search for courses, streams and subtitles

### Motivation and Context
This Change solves the problem of finding lectures in a large list of lectures. It also enhances the search for subtitles or streams in courses.
Closes  #1337.

### Description
The change adds a search bar at the top of every page where you can search for courses or contents of a course, depending if you're on a course page or on the home page.

### Steps for Testing
You can't test this change on the Testserver system.

You have to set up your own local instance of TUM-Live for testing and also have to setup your own meilisearch service.
To start the meilisearch service, you can run the following command:
```bash
docker run -it --rm \
    -p 7700:7700 \
    -e MEILI_MASTER_KEY='MASTER_KEY'\
    -v $(pwd)/meili_data:/meili_data \
    getmeili/meilisearch:v1.9 \
    meilisearch --env="development"
```

If the data isn't automatically exported to meilisearch, you can run this job from the ```Maintenance``` Tab in the Admin panel.

You can enter your search query in the search bar on every page or you can press 'Enter' while the search bar is focused to see the full search page with filters etc.
